### ISSUE TITLE
unnecessary-else giving false positives for errors on `else if` after non-returning if

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [palantir/tslint#4600]
### Bug Report

- __TSLint version__: 5.14.0
- __TypeScript version__: 5.4.0
- __Running TSLint via__:  CLI

#### TypeScript code being linted

```ts
function sample(condition) {
    if (condition === 1) {
        console.log("a");
    } else if (condition === 2) {
        return;
    } else if (condition % 2 === 1) {
        console.log("b");
    }

    console.log("c");
}
```

with `tslint.json` configuration:

```json
{
    "rules": {
        "no-unnecessary-else": true
    }
}
```

#### Actual behavior

The second `else if` cannot be removed, as that would change the behavior of this code.
Right now, if given `1`, it logs:

```
> sample(1)
a
c
```

...but if the second `else if` becomes an `if`:

```ts
function sample(condition) {
    if (condition === 1) {
        console.log("a");
    } else if (condition === 2) {
        return;
    }

    if (condition % 2 === 1) {
        console.log("b");
    }

    console.log("c");
}
```

...then the logic changes and `b` is now logged.

```
> sample(1)
a
b
c
```

#### Expected behavior

If any preceding `if` or `else if` block does not end with a jump statement, an `else` or `else if` should not be considered unnecessary.
